tests/test_naive_bayes.py

```diff
@@ -0,0 +1,394 @@
+from __future__ import (division)
+
+from pomegranate import *
+from nose.tools import with_setup
+from nose.tools import assert_almost_equal
+from nose.tools import assert_equal
+from nose.tools import assert_not_equal
+from nose.tools import assert_less_equal
+from nose.tools import assert_raises
+import random
+import numpy as np
+
+def setup_models():
+	"""
+	split into several methods, one for predictions, one for fitting then predictions, one for constructors
+	use setup to create naive bayes models. test to see if distributions are correct
+	"""
+	# create univarite distributions, test values of distributions
+	normal = NormalDistribution( 5, 2 )
+	uniform = UniformDistribution( 0, 10 )
+
+	# make sure univariate distributions are working
+	assert_almost_equal( normal.log_probability( 11 ), -6.112085713764219 )
+	assert_almost_equal( normal.log_probability( 9 ), -3.612085713764219 )
+	assert_almost_equal( normal.log_probability( 7 ), -2.112085713764219 )
+	assert_almost_equal( normal.log_probability( 5 ), -1.612085713764219 )
+	assert_almost_equal( normal.log_probability( 3 ), -2.112085713764219 )
+	assert_almost_equal( normal.log_probability( 1 ), -3.612085713764219 )
+	assert_almost_equal( normal.log_probability( -1 ), -6.112085713764219 )
+
+	assert_almost_equal( uniform.log_probability( 11 ), -float('inf') )
+	assert_almost_equal( uniform.log_probability( 10 ), -2.3025850929940455 )
+	assert_almost_equal( uniform.log_probability( 5 ), -2.3025850929940455 )
+	assert_almost_equal( uniform.log_probability( 0 ), -2.3025850929940455 )
+	assert_almost_equal( uniform.log_probability( -1 ), -float('inf') )
+	
+	# create multivariate distributions, test values of distributions
+	multi = MultivariateGaussianDistribution( means=[ 5, 5 ], covariance=[[ 2, 0 ], [ 0, 2 ]] )
+	indie = IndependentComponentsDistribution( distributions=[ UniformDistribution(0, 10), UniformDistribution(0, 10) ])
+
+	# make sure multivariate distributions are working
+	assert_almost_equal( multi.log_probability([ 11, 11 ]), -20.531024246969945 )
+	assert_almost_equal( multi.log_probability([ 9, 9 ]), -10.531024246969945 )
+	assert_almost_equal( multi.log_probability([ 7, 7 ]), -4.531024246969945 )
+	assert_almost_equal( multi.log_probability([ 5, 5 ]), -2.5310242469699453 )
+	assert_almost_equal( multi.log_probability([ 3, 3 ]), -4.531024246969945 )
+	assert_almost_equal( multi.log_probability([ 1, 1 ]), -10.531024246969945 )
+	assert_almost_equal( multi.log_probability([ -1, -1 ]), -20.531024246969945 )
+
+	assert_almost_equal( indie.log_probability([ 11, 11 ]), -float('inf') )
+	assert_almost_equal( indie.log_probability([ 10, 10 ]), -4.605170185988091 )
+	assert_almost_equal( indie.log_probability([ 5, 5 ]), -4.605170185988091 )
+	assert_almost_equal( indie.log_probability([ 0, 0 ]), -4.605170185988091 )
+	assert_almost_equal( indie.log_probability([ -1, -1 ]), -float('inf') )
```

Comment:
put these in their own function too. test_univariate_nb and test_multivariate_nb or something